plugins/CoreHome/Controller.php

```diff
@@ -185,10 +187,44 @@ protected function setDateTodayIfWebsiteCreatedToday()
     public function index()
     {
         $this->setDateTodayIfWebsiteCreatedToday();
+
+        if ($this->noRecordedData()) {
+            return $this->showSiteNoData();
+        }
+
         $view = $this->getDefaultIndexView();
         return $view->render();
     }
 
+    private function noRecordedData()
+    {
+        $sql = "SELECT COUNT(*)
+                FROM " . Common::prefixTable('log_visit') . "
+                WHERE idsite=?";
+
+        $count = Db::fetchOne($sql, array($this->idSite));
+
+        return $count == 0;
+    }
```

Comment:
Should this maybe all (Controller etc) be in the SitesManager plugin that listens to the `Controller.CoreHome.index` event? Not sure if that works. There could be also an event listener in the SitesManager to `Request.getRenamedModuleAndAction` and rename `module` and `action` if needed (if it is `CoreHome.index` and has no tracking request). In theory CoreHome does not know anything about tracking code etc.

This SQL could go into a separate class eg in `Piwik\Tracker\Model` but could be also a class in the SitesManager plugin etc.

Maybe also apply a `Limit 1` to the query as we only want to know whether there was at least one tracking request.